autogluon/scheduler/fifo.py

```diff
@@ -257,13 +258,17 @@ def _run_reporter(self, task, task_job, reporter, searcher):
         last_result = None
         while not task_job.done():
             reported_result = reporter.fetch()
+            if reported_result is None:
```

Comment:
@zhanghang1989 this is the while loop, which will continue calling `queue.get` untill it will get something from there or `task_job.done() is True`

Reply:
@zhanghang1989 It comes down to whether a call of reporter.fetch() is blocking, in the sense it waits until fetch() returns something. I kind assumed this was the case.

If the call is not blocking, our current code is broken, because it would try to add a signal from the reporter even if fetch() returned an empty dict (I think this is the issue, not so much None).

Reply:
The reason I assumed that reporter.fetch() is blocking is that I see reporter.move_on() afterwards. If the reporter would simply just continue, why we would need that call?

Reply:
@mseeger Thanks for reviewing the PR.` reporter.fetch()` is a blocking call. This PR does break our code.